introduction_to_applying_machine_learning/deepar_chicago_traffic_violations/deepar_chicago_traffic_violations.ipynb

```diff
@@ -0,0 +1,467 @@
+    "                                       output_path='s3://{}/{}'.format(bucket, s3_output_path),\n",
+    "                                       sagemaker_session=sess)\n",
+    "freq = 'D'\n",
+    "context_length = 30\n",
+    "\n",
+    "deepar.set_hyperparameters(time_freq=freq,\n",
+    "                           context_length=str(context_length),\n",
+    "                           prediction_length=str(prediction_length))\n",
+    "\n",
+    "hyperparameter_ranges = {'mini_batch_size': IntegerParameter(100, 400),\n",
+    "                         'epochs': IntegerParameter(200, 400),\n",
+    "                         'num_cells': IntegerParameter(30,100),\n",
+    "                         'learning_rate': ContinuousParameter(0.0001, 0.1)}\n",
+    "\n",
+    "objective_metric_name = 'test:RMSE'\n",
+    "\n",
+    "tuner = HyperparameterTuner(deepar,\n",
+    "                            objective_metric_name,\n",
+    "                            hyperparameter_ranges,\n",
+    "                            max_jobs=10,\n",
+    "                            strategy='Bayesian',\n",
+    "                            objective_type='Minimize',\n",
+    "                            max_parallel_jobs=10,\n",
+    "                            early_stopping_type='Auto')\n",
+    "\n",
+    "s3_input_train = sagemaker.s3_input(s3_data='s3://{}/{}/train/'.format(bucket, prefix),\n",
+    "                                    content_type='json')\n",
+    "s3_input_test = sagemaker.s3_input(s3_data='s3://{}/{}/test/'.format(bucket, prefix),\n",
+    "                                   content_type='json')\n",
+    "\n",
+    "tuner.fit({'train': s3_input_train, 'test': s3_input_test}, \n",
+    "          include_cls_metadata=False)\n",
+    "job_name = tuner.latest_tuning_job.name\n",
+    "tuner.wait()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "## Host\n",
+    "\n",
+    "We use the [HyperParameterTuner](https://sagemaker.readthedocs.io/en/stable/tuner.html) to host the best model using a single ml.m4.xlarge instance."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "best_tuning_job_name = tuner.best_training_job()\n",
+    "endpoint_name = tuner.deploy(initial_instance_count=1,\n",
+    "                             endpoint_name=best_tuning_job_name,\n",
+    "                             instance_type='ml.m4.xlarge',\n",
+    "                             wait=True)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Evaluate\n",
+    "\n",
+    "To evaluate the model, define a DeepARPredictor class. This class extends the [RealTimePredictor](https://sagemaker.readthedocs.io/en/stable/predictors.html) class. Implementing encode and decode function helps making requests using `pandas.Series` objects rather than raw JSON strings.\n",
```

Comment:
* "define" --> "we define"
* "Implementing encode and decode function helps making" --> "Implementing encode and decode functions helps us make"

Reply:
fixed